stories/src/multi-drag/utils.js

```diff
@@ -0,0 +1,266 @@
+    const destinationIndexOffset: number = selectedTaskIds.reduce(
+      (previous: number, current: TaskId): number => {
+        if (current === dragged) {
+          return previous;
+        }
+
+        const final: Column = entities.columns[destination.droppableId];
+        const column: Column = getHomeColumn(entities, current);
+
+        if (column !== final) {
+          return previous;
+        }
+
+        const index: number = column.taskIds.indexOf(current);
+
+        if (index >= destination.index) {
+          return previous;
+        }
+
+        // the selected item is before the destination index
+        // we need to account for this when inserting into the new location
+        return previous + 1;
+      }, 0);
+
+    const result: number = destination.index - destinationIndexOffset;
+    return result;
+  })();
+
+  // doing the ordering now as we are required to look up columns
+  // and know original ordering
+  const orderedSelectedTaskIds: TaskId[] = [...selectedTaskIds];
+  orderedSelectedTaskIds.sort((a: TaskId, b: TaskId): number => {
+    // moving the dragged item to the top of the list
+    if (a === dragged) {
+      return -1;
+    }
+    if (b === dragged) {
+      return 1;
+    }
+
+    // sorting by their natural indexes
+    const columnForA: Column = getHomeColumn(entities, a);
+    const indexOfA: number = columnForA.taskIds.indexOf(a);
+    const columnForB: Column = getHomeColumn(entities, b);
+    const indexOfB: number = columnForB.taskIds.indexOf(b);
+
+    if (indexOfA !== indexOfB) {
+      return indexOfA - indexOfB;
+    }
+
+    // sorting by their order in the selectedTaskIds list
+    return -1;
+  });
+
+  // we need to remove all of the selected tasks from their columns
+  const withRemovedTasks: ColumnMap = entities.columnOrder.reduce(
+    (previous: ColumnMap, columnId: Id): ColumnMap => {
+      const column: Column = entities.columns[columnId];
+
+      // remove the id's of the items that are selected
+      const remainingTaskIds: TaskId[] = column.taskIds.filter(
+        (id: TaskId): boolean => !selectedTaskIds.includes(id)
+      );
+
+      previous[column.id] = withNewTaskIds(column, remainingTaskIds);
+      return previous;
+    }, entities.columns);
+
+  const final: Column = withRemovedTasks[destination.droppableId];
+  const withInserted: TaskId[] = (() => {
+    const base: TaskId[] = [...final.taskIds];
+    base.splice(insertAtIndex, 0, ...orderedSelectedTaskIds);
+    return base;
+  })();
+
+  // insert all selected tasks into final column
+  const withAddedTasks: ColumnMap = {
+    ...withRemovedTasks,
+    [final.id]: withNewTaskIds(final, withInserted),
+  };
+
+  // $ExpectError - using spread
+  const updated: Entities = {
+    ...entities,
+    columns: withAddedTasks,
+  };
+
+  return {
+    entities: updated,
+    selectedTaskIds: orderedSelectedTaskIds,
+  };
+};
+
+export const reorderBetweenColumns = (args: Args): Result => {
```

Comment:
Doesn't this function reorder tasks in the same column as well? Is there a better name for this?

Reply:
Hmm!

Reply:
`mutliDragAwareReorder` seems alright for now